pallets/subtensor/src/registration.rs

```diff
@@ -169,6 +172,101 @@ impl<T: Config> Pallet<T> {
         Ok(())
     }
 
+    // ---- The implementation for the extrinsic associate: associate a coldkey and a hotkey without registration
+    //
+    // # Args:
+    // 	* 'origin': (<T as frame_system::Config>RuntimeOrigin):
+    // 		- The signature of the calling coldkey. 
+	//
+    // 	* 'hotkey' ( T::AccountId ):
+    // 		- Hotkey to be associated with origin coldkey.
+    //   
+    // # Event:
+    // 	* HotkeyAssociated;
+    // 		- On successfully associating a hotkey with the origin coldkey.
+    //
+    // # Raises:
+	// 	* 'AlreadyRegistered':
+    // 		- The hotkey is already associated with a coldkey.
+    //
+
+    pub fn do_associate( 
+        origin: T::RuntimeOrigin,
+        hotkey: T::AccountId,
+		sig: T::Signature
+    ) -> DispatchResult {
+        // --- 1. Check that the caller has signed the transaction. (the coldkey of the pairing)
+        let coldkey = ensure_signed( origin )?; 
+        log::info!("do_associate( coldkey:{:?} hotkey:{:?} )", coldkey, hotkey );
+
+		// --- 2. Check if the coldkey actually has possession of the hotkey.
+
+		// note(rusty): delegate this into a util function so we can pass (accountid, data, sig) -> bool
+		/* Hotkey bytes so we can cast to PublicKey */
+		let hotkey_pubkey: MultiAddress<T::AccountId, ()> = MultiAddress::Id( hotkey.clone() );
+		let binding = hotkey_pubkey.encode();
+
+		// Skip extra 0th byte.
+		let hotkey_bytes: &[u8; 32] = binding[1..].as_ref().try_into().unwrap();
+		let hotkey_sr25519 = sp_core::sr25519::Public(*hotkey_bytes);
```

Comment:
Do we have to make it bytes here first? Can't go straight from `AccountId`?

Reply:
Public has no TryFrom implementation for U256, only H256